packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -261,6 +261,35 @@ pub(crate) async fn register_indexer_assets(
     if let Some(mut multipart) = multipart {
         queries::start_transaction(&mut conn).await?;
 
+        if config.replace_indexer {
+            // Check that at least one indexer exists
+            if (queries::get_indexer_id(&mut conn, &namespace, &identifier).await).is_ok()
+            {
+                tracing::info!(
+                    "--replace-indexer enabled. Removing existing indexer {}.{}",
+                    &namespace,
+                    &identifier
+                );
+
+                if let Err(e) =
+                    queries::remove_indexers(&mut conn, &namespace, &identifier, None)
+                        .await
+                {
+                    error!("Failed to remove Indexer({namespace}.{identifier}): {e}");
+                    queries::revert_transaction(&mut conn).await?;
+                    return Err(e.into());
+                }
+
+                if let Err(e) =
+                    queries::remove_graph(&mut conn, &namespace, &identifier).await
```

Comment:
- This gets to my "single transaction" comment above ☝🏽 
- I think `remove_graph` should be nested inside `remove_indexer`. 
  - There is never a case in which we want to call `remove_graph` without calling `remove_indexer
  - And if its nested, that means we can use a single transaction for the entire removal